doc/api/assert.md

```diff
@@ -257,18 +257,26 @@ property set equal to the value of the `message` parameter. If the `message`
 parameter is undefined, a default error message is assigned.
 
 ## assert.fail(message)
-## assert.fail(actual, expected, message, operator)
+## assert.fail(actual, expected[, message[, operator[, stackStartFunction]]])
 <!-- YAML
 added: v0.1.21
 -->
 * `actual` {any}
 * `expected` {any}
 * `message` {any}
 * `operator` {string} (default: '!=')
+* `stackStartFunction` {function} (default: assert.fail)
 
 Throws an `AssertionError`. If `message` is falsy, the error message is set as
 the values of `actual` and `expected` separated by the provided `operator`.
 Otherwise, the error message is the value of `message`.
+If you provide exactly the two arguments `actual` and `expected`, `operator`
+will default to `'!='`.
+You should either use the `message` or the `operator` but not both at the same
```

Comment:
If we do want to document this quirk, an example demonstrating what does not work would be nice..